code/atom.dm

```diff
@@ -397,6 +397,14 @@
 	var/throw_range = 7
 	var/throwforce = 1
 
+	// Minimap Icons.
+	///Which minimap types this object should display a marker on.
+	var/minimaps_to_display_on
+	///Whether this object is able to move, thus requiring a tracked minimap marker.
+	var/tracked_minimap_marker = FALSE
```

Comment:
This is on atom/movable so isn't anything able to move? Or is this intended to be more explicit than `anchored`? Otherwise it seems redundant.

Reply:
As all objects and mobs are derived from `atom/movable`, but not all objects or mobs are in practice capable of motion, I felt that distinguishing between them was necessary. Consider `/obj/machinery/bot/secbot/beepsky` and `/obj/warp_beacon`; both technically capable of motion, but in practice only one moves. `anchored` could not be used to determine this, as some objects or mobs toggle their anchored state. I agree that the code doc could be more explicit on that, however.
Alternatively, I could have every map marker be a tracking map marker, and accept that immobile atoms will never call `handle_move`, although I'd need further feedback on that.

I'm unsure about your proposal of the vars and proc on `atom/movable` being moved to a component as I am unfamiliar with coding them.